app/Http/Controllers/ManageOrdersController.php

```diff
@@ -98,23 +144,86 @@ public function cancelOrder(string $id)
             ]);
     }
 
+    public function deleteOrderLine(Request $request, string $id)
+    {
+        $validated = $request->validate;
+
+        $orderLine = OrderLine::find($id);
+        if ($orderLine === null) {
+            return redirect()->back()
+                ->with([
+                    'toast-type' => 'error',
+                    'toast-message' => __('manage-orders/order.product-remove-fail')
+                ]);
+        }
+
+        $orderLine->delete();
+
+        return redirect()->back()
+            ->with([
+                'toast-type' => 'success',
+                'toast-message' => __('manage-orders/order.product-removed')
+            ]);
+    }
+
+    public function addProduct(Request $request, string $id)
+    {
+        $validated = $request->validate([
+            'product-select' => 'required',
+        ]);
+
+        $products = $this->getAllProducts();
+        $product = $products->where('row_number', $validated['product-select'])->first();
+
+        $order = Order::find($id);
+        if ($order === null) {
+            return redirect()->back()
+                ->with([
+                    'toast-type' => 'error',
+                    'toast-message' => __('manage-orders/order.order-doesnt-exist')
+                ]);
+        }
+
+        if ($product === null) {
+            return redirect()->back()
+                ->with([
+                    'toast-type' => 'error',
+                    'toast-message' => __('manage-orders/order.product-add-fail')
+                ]);
+        }
+
+        $orderLine = new OrderLine();
```

Comment:
Je kan hier ook OrderLine::create([...]); doen, is dit zo gedaan voor een reden?

Reply:
Nee niet bewust haha, ik ga t aanpassen!